gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -642,15 +633,15 @@ private void beginResumingDag(DagId dagIdToResume) throws IOException {
      */
     private void finishResumingDags() throws IOException {
       for (Map.Entry<String, Dag<JobExecutionPlan>> dag : this.resumingDags.entrySet()) {
-        JobStatus flowStatus = pollFlowStatus(dag.getValue());
-        if (flowStatus == null || !flowStatus.getEventName().equals(PENDING_RESUME.name())) {
+        Optional<JobStatus> flowStatus = DagManagerUtils.pollFlowStatus(dag.getValue(), this.jobStatusRetriever, this.jobStatusPolledTimer);
+        if (!flowStatus.toJavaUtil().filter(fs -> fs.getEventName().equals(PENDING_RESUME.name())).isPresent()) {
           continue;
         }
 
         boolean dagReady = true;
         for (DagNode<JobExecutionPlan> node : dag.getValue().getNodes()) {
-          JobStatus jobStatus = pollJobStatus(node);
-          if (jobStatus == null || jobStatus.getEventName().equals(FAILED.name()) || jobStatus.getEventName().equals(CANCELLED.name())) {
+          Optional<JobStatus> jobStatus = DagManagerUtils.pollJobStatus(node, this.jobStatusRetriever, this.jobStatusPolledTimer);
+          if (!jobStatus.isPresent() || jobStatus.get().getEventName().equals(FAILED.name()) || jobStatus.get().getEventName().equals(CANCELLED.name())) {
```

Comment:
separately checking `isPresent()` and then `.get()` over a boolean short-circuit is not as common as the single expression:
```
if (jobStatus.filter(js -> {
  String jobName = js.getEventName();
  return jobName.equals(FAILED.name()) || jobName.equals(CANCELLED.name());
  }).isPresent())
```
what do you think?